libs/parallelism/async_base/include/hpx/async_base/launch_policy.hpp

```diff
@@ -138,90 +172,235 @@ namespace hpx {
                 return this->policy_holder_base::is_valid();
             }
 
-            constexpr launch_policy policy() const
+            constexpr launch_policy policy() const noexcept
             {
                 return this->policy_holder_base::get_policy();
             }
-            constexpr threads::thread_priority priority() const
+            constexpr threads::thread_priority priority() const noexcept
             {
                 return this->policy_holder_base::get_priority();
             }
+            constexpr threads::thread_stacksize stacksize() const noexcept
+            {
+                return this->policy_holder_base::get_stacksize();
+            }
+            constexpr threads::thread_schedule_hint hint() const noexcept
+            {
+                return this->policy_holder_base::get_hint();
+            }
         };
 
         ///////////////////////////////////////////////////////////////////////
         struct async_policy : policy_holder<async_policy>
         {
             constexpr explicit async_policy(
                 threads::thread_priority priority =
-                    threads::thread_priority::default_) noexcept
-              : policy_holder<async_policy>(launch_policy::async, priority)
+                    threads::thread_priority::default_,
+                threads::thread_stacksize stacksize =
+                    threads::thread_stacksize::default_,
+                threads::thread_schedule_hint hint = {}) noexcept
+              : policy_holder<async_policy>(
+                    launch_policy::async, priority, stacksize, hint)
             {
             }
 
-            constexpr async_policy operator()(
-                threads::thread_priority priority) const noexcept
+            constexpr async_policy operator()(threads::thread_priority priority,
+                threads::thread_stacksize stacksize =
+                    threads::thread_stacksize::default_,
+                threads::thread_schedule_hint hint = {}) const noexcept
             {
-                return async_policy(priority);
+                return async_policy(priority, stacksize, hint);
+            }
+
+            async_policy& operator=(threads::thread_priority priority) noexcept
+            {
+                this->priority_ = priority;
+                return *this;
+            }
```

Comment:
I find this a bit questionable. I think I'd expect
```
policy = priority;
```
to behave the same as
```
policy = {priority};
```
If anything maybe
```
policy += priority;
```
could change only the priority, but even that is very questionable.

Should we not use `with_priority/stacksize/hint` as we do with the executors instead?

Reply:
Yes, I was already planning to change this. I will replace the `operator=()` with equivalent `operator()()` overloads, which will make things more `constexpr` friendly as well.

Reply:
Do you mean yop'd change it to have
```
policy(priority);
```
or, possibly return a new policy:
```
policy = policy(priority);
```
?

As I said, why not `with_x` as we do elsewhere?

Reply:
> Do you mean yop'd change it to have
> 
> ```
> policy(priority);
> ```
> 
> or, possibly return a new policy:
> 
> ```
> policy = policy(priority);
> ```
> 
> ?

yes

> As I said, why not `with_x` as we do elsewhere?

Isn't that a bit overkill?

Reply:
Is it overkill for executors? If no, I don't think it is for launch policies either. If this is indeed meant to as a user-facing facility (which it is, right?) then I'd rather not have to explain to users that to set a priority for an executor you use `with_priority(exec, prio)` but for launch policies you do `policy(prio)`.

Reply:
> yes, even more as we've already had `launch::async(priority)`

we also have `parallel_policy_executor(priority)` so should have `exec(priority)` as well? I think no. Those are constructors where it's much more explicit that you're creating a new instance. Overloading the call operator for this purpose is more confusing than helpful, to me at least. Plus it has the downside that getting the priority (or any property like that) still needs a plain member function which is to me a weirdly asymmetric situation.

Reply:
> Is it overkill for executors? If no, I don't think it is for launch policies either. If this is indeed meant to as a user-facing facility (which it is, right?) then I'd rather not have to explain to users that to set a priority for an executor you use `with_priority(exec, prio)` but for launch policies you do `policy(prio)`.

Fair point. Should we deprecate `hpx::launch::async(priority)` in this case?

Reply:
> Fair point. Should we deprecate `hpx::launch::async(priority)` in this case?

I don't think that's necessary. It's still useful just like the constructors for the parallel executor are useful. I was mainly concerned with having redundant ways of setting and getting those properties.